src/components/controllers/page/index.js

```diff
@@ -161,6 +161,17 @@ export default function Page({ route, prev, next }, ctx) {
 	const canEdit = showTitle && current != '404' && current !== '/blog';
 	const hasSidebar = meta.toc !== false && isDocPage(url);
 
+	useEffect(() => {
+		if (location.hash) {
+			const anchor = document.querySelector(location.hash);
+			if (anchor) {
+				// Do not use scrollIntoView as it will cause
+				// the heading to be covered by the header
+				scrollTo({ top: anchor.offsetTop });
```

Comment:
```suggestion
				scrollTo(0, anchor.getBoundingClientRect().top);
```

Reply:
Looking at this again, `getBoundingClientRect().top` would get the top of the wrapping element and would need to be inverted (that will return a negative when off the bottom of the screen).